File: frontend/ui_tests/pages/DashboardPage.ts
import { Page, expect } from '@playwright/test';

export class DashboardPage {
  constructor(private page: Page) {}

  async open() {
    await this.page.goto('/dashboard');
  }

  async clickApplyFilters() {
    await this.page.getByTestId('apply-filters').click();
  }

  async expectInvoiceTableVisible() {
    await expect(
      this.page.getByTestId('invoice-table')
    ).toBeVisible();
  }
}

File: frontend/ui_tests_python/dashboard_page.py
class DashboardPage:
    def __init__(self, page):
        self.page = page

    def open(self, base_url):
        self.page.goto(f"{base_url}/dashboard")

    def apply_filters(self):
        self.page.get_by_test_id("apply-filters").click()

    def invoice_table(self):
        return self.page.get_by_test_id("invoice-table")
